components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java

```diff
@@ -210,5 +215,22 @@ protected void unsetUserStoreConfigListenerService(UserStoreConfigListener userS
         UserStoreConfigListenersHolder.getInstance().unsetUserStoreConfigListenerService(userStoreConfigListener);
     }
 
+    @Reference(
+            name = "user.store.config.service",
+            service = UserStoreConfigService.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetUserStoreConfigService"
+    )
+    protected void setUserStoreConfigService(UserStoreConfigService userStoreConfigService) {
```

Comment:
since your component itself register this service, you can call `UserStoreConfigListenersHolder.getInstance().setUserStoreConfigService(userStoreConfigService);` from the place where you register the service in the activate method.